tutorials/firefly/OpenUniverse2024Preview_Firefly.md

```diff
@@ -0,0 +1,750 @@
+axs[1].set_title(f"RUBIN in filter {filter_rubin}")
+
+fig.suptitle(f"Cutouts at ({coord.ra}, {coord.dec}) with {cutout_size} size", fontsize=14)
+plt.tight_layout(rect=[0, 0, 1, 0.97])
+```
+
+## Use Firefly to interactively identify a blended source
+
+Clearly, the simulated Roman coadd has higher spatial resolution than the Rubin simulated coadd. Let's try to locate blended objects to compare in the simulated Rubin and Roman images. We will use Firefly's interactive visualization to make this task easier.
+
++++
+
+### Launch and initialize Firefly
+There are two ways to initialize a Firefly client from Python, depending on whether you're running the notebook in JupyterLab or not. Assuming you have `jupyter-firefly-extensions` set up in your environment as explained [here](https://github.com/Caltech-IPAC/jupyter_firefly_extensions/blob/master/README.md), you can use `make_lab_client()` in JupyterLab, which will open the Firefly viewer in a new tab within the Lab. Otherwise, you can use `make_client()` in a Jupyter Notebook (or even a Python shell), which will open the Firefly viewer in a new web browser tab.
+
+You also need a Firefly server to communicate with your Firefly Python client. In this notebook, we use a public Firefly server: the IRSA Viewer (https://irsa.ipac.caltech.edu/irsaviewer). However, you can also run a local Firefly server via a [Firefly Docker image](https://hub.docker.com/r/ipac/firefly) and access it at `http://localhost:8080/firefly`. The URL of the Firefly server is read by both `make_client()` and `make_lab_client()` through the environment variable `FIREFLY_URL`. However, `make_client()` also allows you to pass the URL directly as the `url` parameter.
+
+```{code-cell} ipython3
+# Uncomment when using within Jupyter Lab with jupyter_firefly_extensions installed
+# fc = FireflyClient.make_lab_client()
+
+# Uncomment for contexts other than above 
+fc = FireflyClient.make_client(url="https://irsa.ipac.caltech.edu/irsaviewer")
+
+fc.reinit_viewer() # to clean the state, if this cell ran earlier
+```
+
+### Send the simulated Rubin coadd to Firefly using show_fits.
+
+For more information on `show_fits`, see: 
+
+https://caltech-ipac.github.io/firefly_client/api/firefly_client.FireflyClient.html#firefly_client.FireflyClient.show_fits
+
+```{code-cell} ipython3
+coadd_ff_id_rubin = 'rubin-coadd-filter-r'
+fc.show_fits(url=https_url(coadd_s3_fpath_rubin),
+             plot_id=coadd_ff_id_rubin,
+             Title="Rubin Coadd"
+             )
+```
+
+### Use ds9 region syntax to overplot the simulated Roman image blocks on the interactive display
+
+The Firefly client includes several methods related to controlling ds9 region overlays. To 
+overlay a region layer on the loaded FITS images, we can use `overlay_region_layer`:
+
+https://caltech-ipac.github.io/firefly_client/api/firefly_client.FireflyClient.html#firefly_client.FireflyClient.overlay_region_layer
```

Comment:
these look really ugly, please make them a link instead with a shorter text. 

(I'm not certain that interpshinx linking would work out of the box, we can open an issue for it and I can look into it in the next year; for now manually linking these is  way to go)

Reply:
Agreed, they all are looking ugly. I thought @vandesai1 kept it in this formatting so that audience can know and click it. I can change it to hyperlinked method name?

Reply:
yes, I think hyperlinked method, or class+method name is the way to go, just like as it would look like in a sphinx API docs